### DISCUSSION TITLE
How to customize current_user dependency to add a custom logic?

### DISCUSSION THREAD

**Discussion by Tur8008** [fastapi-users/fastapi-users#1242] [Q&A] [answered]
Hello!
I have an admin site in my project and I need to prevent the users that are not staff from getting access to admin site's pages.
So I added is_staff boolean field to User model and I just need to add some additional logic to `fastapi_users.current_user` dependency callable.
I need to add something like that:
```python
if user.is_superuser or user.is_staff:
# authenticate user
else:
# return 403 Forbiden
```
How to implement it ?

**Comment by frankie567** [comment, marked as answer]
You can wrap the dependency with your own. That's what nice with FastAPI: dependencies can recursively call dependencies!

```py
async def current_staff_user(user = Depends(current_active_user)):
    if user.is_superuser or user.is_staff:
        return user
    raise HTTPException(status_code=403)
```

**Comment by Tur8008** [comment]
Thank you!
